### DISCUSSION TITLE
How to work with apps/ specific environment variables?

### DISCUSSION THREAD

**Discussion by rwieruch** [vercel/turborepo#3867] [Help] [answered]
I am working with the following monorepo structure:

```
- apps/
--- app-one/
-----.env.foo
-----.env.bar
--- app-two/
-----.env.foo
-----.env.zoo
--- docs/
- packages/
--- package-one/
```

While app-one/ and app-two/ are Vite projects where I can call the package.json script with a specific .env file when performing `yarn dev --mode foo`, the docs/ project is a Storybook application which does not allow the --mode flag. Furthermore, the package-one/ does not allow this flag too.

My global package.json has a `turbo run dev` script. Now I could pass the mode from there via `turbo run dev -- -- --mode foo`, however, some of the apps/ and packages/ will fail because they do not accept this argument. Is there any way to just ignore arguments if they are not working or only pass the argument to certain applications in apps/?

**Comment by anthonyshew** [comment, marked as answer]
We've published some documentation around environment variables best practices here: https://turbo.build/repo/docs/crafting-your-repository/using-environment-variables#best-practices
